### ISSUE TITLE
Opening being greened.

### ISSUE THREAD

**Issue by Chesterxx** [hilts-vaughan/UnlinkMKV-GUI#9]
I'm new to segmenting, and i'm not sure what it's called, but this is the issue i'm having.

I was unlinking Kiss x Sis (Coalgirls release from BakaBT). The majority of the first episodes turned out fine. The 11th and 12th episode had an issue where the opening audio and subtitles were fine, but the video was green.

Example: http://puu.sh/jGZIJ/9c089d533a.png

The endings seem fine. I have the latest version of all requirements (MKVMerge, UnlinkMKV-GUI and req etc.

http://pastebin.com/2fqx114j

I have tested the MKV on both VLC and Windows Media Player.

Any ideas? :(

**Comment by hilts-vaughan** [comment]
Hi,

Thanks for the report. I'll take a look for you.. I wonder if something
strange is going on.. codec issue maybe? If you have another device, try
playing it there.

I'm going to try it on mpv tonight on Linux. I'll keep you posted.

**Comment by Chesterxx** [comment]
Okay thanks mate, i'll go ahead and test it across my devices and come back to you. If theres anything else i can do just say.

**Comment by Chesterxx** [comment]
Plays with the same result across my devices.

**Comment by hilts-vaughan** [comment]
Must be an encoding problem. Thanks, I'll try and patch this today for you.

**Comment by Chesterxx** [comment]
Just wanted to check up, was there any development with the issue? Did you find out what it is that is doing this?

**Comment by hilts-vaughan** [comment]
Confirmed broken. Working on a fix.

**Comment by shogunreaper** [comment]
so this was never fixed?

was trying to use this on some of 10bit mkvs and it merged them (always saying "fixing subs, again... (maybe an mkvmerge issue?)"

heres a mediainfo of one of the files.

`Format                      : Matroska
Format version              : Version 4 / Version 2
File size                   : 130 MiB
Duration                    : 23mn 44s
Overall bit rate            : 765 Kbps
Movie name                  : (Hi10)_Rakudai_Kishi_no_Cavalry_-_01_(720p)_(Doki)
Encoded date                : UTC 2016-06-09 04:29:35
Writing application         : mkvmerge v6.1.0 ('Old Devil') built on Mar  4 2013 09:10:42
Writing library             : libebml v1.3.0 + libmatroska v1.4.0
Attachments                 : Alegreya-Regular.otf / AlegreyaSans-Regular.otf / AlegreyaSubSans-Bold.otf / ArnoPro-BoldDisplay.otf / ArnoPro-Display.otf / ArnoPro-SmbdDisplay.otf / CherrySwash-Bold.otf / HelveticaNeueLTStd-Lt.otf / HelveticaNeueLTStd-LtEx.otf / VestaPro-Bold.otf / VestaPro-BoldItalic.otf / arial.ttf / BRIANNE_S_HAND.TTF / HelveticaNeue.ttf / kaiu.ttf / Niew CroMagnon.ttf / pala.ttf / SPLAT.TTF / PROFILEREGULAR.TTF

Video
ID                          : 1
Format                      : AVC
Format/Info                 : Advanced Video Codec
Format profile              : High 4:4:4 Predictive@L5
Format settings, CABAC      : Yes
Format settings, ReFrames   : 12 frames
Codec ID                    : V_MPEG4/ISO/AVC
Duration                    : 23mn 44s
Width                       : 1 280 pixels
Height                      : 720 pixels
Display aspect ratio        : 16:9
Frame rate mode             : Constant
Frame rate                  : 23.976 (24000/1001) fps
Color space                 : YUV
Chroma subsampling          : 4:4:4
Bit depth                   : 10 bits
Scan type                   : Progressive
Writing library             : x264 core 148 r2638kMod 7599210
Encoding settings           : cabac=1 / ref=12 / deblock=1:1:1 / analyse=0x3:0x133 / me=umh / subme=10 / psy=1 / fade_compensate=0.00 / psy_rd=0.70:0.00 / mixed_ref=1 / me_range=32 / chroma_me=1 / trellis=2 / 8x8dct=1 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=4 / threads=6 / lookahead_threads=1 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / constrained_intra=0 / fgo=0 / bframes=10 / b_pyramid=2 / b_adapt=2 / b_bias=0 / direct=3 / weightb=1 / open_gop=0 / weightp=2 / keyint=250 / keyint_min=23 / scenecut=45 / intra_refresh=0 / rc_lookahead=60 / rc=crf / mbtree=1 / crf=23.0000 / qcomp=0.60 / qpmin=0 / qpmax=81 / qpstep=4 / ip_ratio=1.40 / aq=2:1.00
Language                    : English
Default                     : Yes
Forced                      : No
Statistics Tags Issue       : mkvmerge v8.0.0 ('Til The Day That I Die') 32bit 2015-11-10 03:04:07 / mkvmerge v6.1.0 ('Old Devil') built on Mar  4 2013 09:10:42 2016-06-09 04:29:35
FromStats_BitRate           : 566454
FromStats_Duration          : 00:22:14.250000000
FromStats_FrameCount        : 31990
FromStats_StreamSize        : 94473971

Audio
ID                          : 2
Format                      : AAC
Format/Info                 : Advanced Audio Codec
Format profile              : LC
Codec ID                    : A_AAC
Duration                    : 23mn 44s
Channel(s)                  : 2 channels
Channel positions           : Front: L R
Sampling rate               : 48.0 KHz
Frame rate                  : 46.875 fps (1024 spf)
Compression mode            : Lossy
Delay relative to video     : 20ms
Title                       : Japanese (AAC)
Language                    : Japanese
Default                     : Yes
Forced                      : No
Statistics Tags Issue       : mkvmerge v8.0.0 ('Til The Day That I Die') 32bit 2015-11-10 03:04:07 / mkvmerge v6.1.0 ('Old Devil') built on Mar  4 2013 09:10:42 2016-06-09 04:29:35
FromStats_BitRate           : 69014
FromStats_Duration          : 00:22:14.250000000
FromStats_FrameCount        : 62543
FromStats_StreamSize        : 11510319

Text
ID                          : 3
Format                      : ASS
Codec ID                    : S_TEXT/ASS
Codec ID/Info               : Advanced Sub Station Alpha
Compression mode            : Lossless
Title                       : Doki (English)
Language                    : English
Default                     : Yes
Forced                      : No
Statistics Tags Issue       : mkvmerge v8.0.0 ('Til The Day That I Die') 32bit 2015-11-10 03:04:07 / mkvmerge v6.1.0 ('Old Devil') built on Mar  4 2013 09:10:42 2016-06-09 04:29:35
FromStats_BitRate           : 1712
FromStats_Duration          : 00:22:12.070000000
FromStats_FrameCount        : 816
FromStats_StreamSize        : 285118

Menu
00:00:00.000                : en:Prologue
00:02:56.968                : en:Part A
00:16:24.942                : en:Part B
00:22:09.202                : en:Opening
00:23:39.209                : en:Preview`

**Comment by hilts-vaughan** [comment]
It's fixed for some files. There's a bug in newer versions.

Can you send me an email with the filenames in question? More samples to
inspect won't hurt.

**Comment by shogunreaper** [comment]
just the filenames or you want a link to the torrent with the files?

**Comment by hilts-vaughan** [comment]
Whatever you're using to get the files is fine. I'd like to investigate on
a few different versions.

**Comment by shogunreaper** [comment]
i got the links from here

http://hi10anime.com/category/rakudai-kishi-no-cavalry/

also happened for this series too:

http://hi10anime.com/jormungand/jormungand-series/

I'm guessing its something to do with their encoding settings.

**Comment by mar789** [comment]
Any update on this? Been having a problem with the second ending on the second half of Gundam ZZ. At this point I think I've tried everything to get it to work.

**Comment by hilts-vaughan** [comment]
It's a bug with encoding. Can you check if the two files are encoded the
same way before merging?

**Comment by mar789** [comment]
As far as I can tell from MediaInfo they are encoded the same. Any differences there are such as the writing library exist in the first opening and ending that was used in the first half of the series which merged perfectly fine. From what I can tell there should be no reason why there is a problem merging.

Edit: Looking deeper the only actual difference I see in the ending is very little of the encoding settings.

This is the ending that doesn't work.

> cabac=1 / ref=3 / deblock=1:0:0 / analyse=0x3:0x113 / me=hex / subme=7 / psy=1 / psy_rd=1.00:0.00 / mixed_ref=1 / me_range=16 / chroma_me=1 / trellis=1 / 8x8dct=1 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=-2 / threads=6 / sliced_threads=0 / nr=0 / decimate=1 / mbaff=0 / constrained_intra=0 / bframes=3 / b_pyramid=2 / b_adapt=1 / b_bias=0 / direct=1 / wpredb=1 / wpredp=2 / keyint=100 / keyint_min=25 / scenecut=40 / intra_refresh=0 / rc_lookahead=40 / rc=crf / mbtree=1 / crf=23.0 / qcomp=0.60 / qpmin=10 / qpmax=51 / qpstep=4 / vbv_maxrate=20000 / vbv_bufsize=25000 / ip_ratio=1.40 / aq=1:1.00  

This is the ending for the first half of the series that does work. 

> cabac=1 / ref=4 / deblock=1:1:1 / analyse=0x3:0x113 / me=umh / subme=9 / psy=1 / psy_rd=0.8:0.0 / mixed_ref=1 / me_range=24 / chroma_me=1 / trellis=1 / 8x8dct=1 / cqm=0 / deadzone=21,11 / chroma_qp_offset=-2 / threads=6 / nr=0 / decimate=1 / mbaff=0 / bframes=3 / b_pyramid=0 / b_adapt=1 / b_bias=0 / direct=2 / wpredb=1 / keyint=240 / keyint_min=25 / scenecut=40 / rc_lookahead=40 / rc=crf / mbtree=1 / crf=17.0 / qcomp=0.60 / qpmin=10 / qpmax=51 / qpstep=4 / vbv_maxrate=50000 / vbv_bufsize=50000 / ip_ratio=1.40 / aq=1:1.00 / pulldown=0 / nal_hrd=0

**Comment by mar789** [comment]
At this point I think I've tried every setting to get this to work, even tried using the command line version but even that returned the same problem. I feel the only way to properly get these openings/endings that get messed up is to do it the manual way by getting the time codes and rebuilding the whole episode yourself.

**Comment by hilts-vaughan** [comment]
I'm tagging a new release that will fix some of these issues, other issues are actually not issues with the merge process, but rather at the source. I tried playing the "Kiss x Kiss" source files in VLC while still linked (since VLC supported linked segments) and the result is the same.

To fix this, I reencoded both files to the same format and then re-ran the process. It's not ideal as it is manual, but it will get around this. 

I will close this for now, we should track individual releases and provide solutions for them on a case-by-case and fix it in the base application where it is possible.
